src/config/config.cc

```diff
@@ -368,12 +371,11 @@ void Config::initFieldCallback() {
     return srv->storage->SetOptionForAllColumnFamilies(TrimRocksDbPrefix(k), v);
   };
   auto set_compression_type_cb = [](Server *srv, [[maybe_unused]] const std::string &k,
-                                    const std::string &v) -> Status {
+                                    [[maybe_unused]] const std::string &v) -> Status {
     if (!srv) return Status::OK();
-
     std::string compression_option;
     for (auto &option : engine::CompressionOptions) {
-      if (option.name == v) {
+      if (option.type == srv->GetConfig()->rocks_db.compression) {
```

Comment:
Just wondering if this change here is just to make the comparison condition more concrete, or it has to be changed to make `nocompression_for_first_levels` work? Thank you!

Reply:
previously it compare value that come to callback. callback was used once when "compression" option set. Now callback call twice - for set "compression" and for set "nocompression_for_first_n_levels", so value may be name of compression or levels count that no need compression.

if i make two separate callback, it will have dublicate code, because this two options closely related.

Reply:
i dont know how make this code better, if you have solution, plz share with me :)
i never write in c++ before

Reply:
This is a bit weird to reuse the same function...@git-hulk do you have any better idea here? There're 2 configs about compression and would set the system once...

Reply:
I have no better solution for now because both two configurations will affect the same rocksdb option. It's weird while compared to other configurations, but it looks reasonable.

Reply:
Should we separate it to:

```
set_compression_method_callback
set_no_compression_level_callback
```

?

Reply:
@mapleFU Do you mean this?

```C++
rocksdb::Status set_rocksdb_compression(CompressionType compression, int nocompression_for_first_n_levels) {
   ...
}

rocksdb::Status set_compression_method_callback() {
   ...
   set_rocksdb_compression(compression, nocompression_for_first_n_levels)
}

rocksdb::Status set_no_compression_level_callback() {
  ...
  set_rocksdb_compression(compression, nocompression_for_first_n_levels)
}

```

If yes, I'm good with it.

Reply:
i'am done in separate commit, but i dont see any benefits. set_db_option_cb used more then 1 times for example